ansible/playbooks/roles/redpanda_broker/tasks/start-redpanda.yml

```diff
@@ -83,13 +147,20 @@
   changed_when: '"New configuration version is {{ config_version.stdout|int() }}." not in result.stdout'
   run_once: true
 
+# see if a cluster config operation should trigger ansible to restart each
+# redpanda instance in the cluster
+#
 - name: Check if restart needed
   ansible.builtin.shell:
     cmd: "rpk cluster config status | grep '^{{ node_id }} ' | awk '{ print $3 }' |  grep -E 'true|false'"
   register: restart_required_rc
   changed_when: False
 
-- name: Generate Node config
+# We need to run the node config again after first run because some things
+# could have changed out-of-band. Since rpk can modify the config on your
+# behalf, we want to reset it back to a known templated state.
```

Comment:
This task will only fire if the playbook is being run on an already initialised cluster, for example, when running the TLS playbooks. So here we're going to set any node properties that we or the user have specified and this will take effect on the next restart.
We can't do it sooner could it could potentially have caused the `rpk cluster config` or `rpk cluster status` commands to have failed

Reply:
Ok, I'll integrate that into the comment. Still seems weird to define this twice even though it's basically doing the same thing. Seems like we'd be better off using a notify on those tasks that should trigger a rebuild of the config, and then stick the config build task in a handler.

Reply:
Ok. I updated comment to reflect the above.